ingest/src/main/java/com/microsoft/azure/kusto/ingest/result/IngestionStatus.java

```diff
@@ -14,20 +17,28 @@
 /// <remarks>
 /// Any change to this class must be made in a backwards/forwards-compatible manner.
 /// </remarks>
-public class IngestionStatus extends TableServiceEntity {
+public class IngestionStatus {
     /// <summary>
     /// The updated status of the ingestion. The ingestion status will be 'Pending'
     /// during the ingestion's process
     /// and will be updated as soon as the ingestion completes.
     /// </summary>
     public OperationStatus status;
+    private Map<String, Object> ingestionInfo = new HashMap<>();
 
     public String getStatus() {
         return status.toString();
     }
 
     public void setStatus(String s) {
-        status = OperationStatus.valueOf(s);
+        if (s != null) {
+            setStatus(OperationStatus.valueOf(s));
```

Comment:
What happens if s is not a valid value? does it throw? put null?

Reply:
enums throws